Blockzilla/BrowserViewController.swift

```diff
@@ -819,7 +820,20 @@ extension BrowserViewController: URLBarDelegate {
     func urlBar(_ urlBar: URLBar, didEnterText text: String) {
         // Hide find in page if the home view is displayed
         let isOnHomeView = homeView != nil
-        overlayView.setSearchQuery(query: text, animated: true, hideFindInPage: isOnHomeView)
+        if Settings.getToggle(.enableSearchSuggestions) {
```

Comment:
I encountered a bug that I think is tied to this line. (bug seen below).

![bug](https://user-images.githubusercontent.com/4400286/47457888-4b241900-d7a6-11e8-928e-0202f044f86d.gif)


When I type one letter, backspace, then type another letter, it looks like the blank getSuggestions is overwriting my next suggestion. In other words:

1. getSuggestions("")
2. getSuggestions("r")
3. getSuggestions("r") returns
4. getSuggestions("") returns (overwriting the first one)

This seems to be fixed by changing this line to:

```if Settings.getToggle(.enableSearchSuggestions) && text != "" { ```

That being said, we should **enforce that no dispatched request for a search suggestion is overwriting the most recent call.** The easiest way I can think to do this is to check whether ```suggestions[0] == text``` before setting the queryArray. If it doesn't match, we should simply discard the result. These two changes in combination should fix this bug!